### ISSUE TITLE
File names pasted in editor when copied from OS file manager.

### ISSUE THREAD

**Issue by Khalilsqu** [spyder-ide/spyder#8566]
So I noticed that when copying a file from OS file manager and pasting it into the editor, its name will be pasted. A similar behaviour is solved with the commit `correct tests and remove text from copy file` in PR #8494 

![name_issue_copy_issue](https://user-images.githubusercontent.com/22060413/51087973-134bc280-1752-11e9-9bea-a84bffbe2da0.gif)

It is not considered a big issue.  But might be preferred to not allow this behaviour to happen.

Spyder: Master (currently under development) 
OS: windows: 10
Python version: 3.6

**Comment by ccordoba12** [comment]
So what should be the expected behavior in this case?

**Comment by Khalilsqu** [comment]
> So what should be the expected behavior in this case?

I Think there should be no text in the clipboard when pasting clipboard data which contains none text contents (mimeData) e.g. files/folders. The text should be set to empty string if the data within the clipboard are mimeData.

**Comment by Khalilsqu** [comment]
@ccordoba12 

By the way. It can be fixed by changing this [function](https://github.com/spyder-ide/spyder/blob/a5ca1f20f69ba8f9bc1b85e166ce865302c06a54/spyder/plugins/editor/widgets/codeeditor.py#L1578) as such.

    ```
@Slot()
    def paste(self):
        """
        Reimplement QPlainTextEdit's method to fix the following issue:
        on Windows, pasted text has only 'LF' EOL chars even if the original
        text has 'CRLF' EOL chars
        """
        clipboard = QApplication.clipboard()
        if clipboard.mimeData().hasUrls():
            clipboard.setText('')
        else:
            text = to_text_string(clipboard.text())
            if len(text.splitlines()) > 1:
                eol_chars = self.get_line_separator()
                text = eol_chars.join((text + eol_chars).splitlines())
                clipboard.setText(text)
            # Standard paste
            TextEditBaseWidget.paste(self)
            self.document_did_change(text)
```

I tested it, This changes does not affect pasting text data (within Spyder editor itself or from outside Spyder) but prevent pasting file/folders names if they are copied in the clipboard from outside Spyder.

**Comment by ccordoba12** [comment]
Ok, I agree with this, except for one thing: instead of using `clipboard.setText('')` we should simply change that line for `return`. That will avoid resetting user's clipboard and pasting its contents in our editor.

What do you think?

**Comment by Khalilsqu** [comment]
Ideal solution, it wont empty users clipboard data in other application if he/she accidentally presss Cntl+V in Spyder editor. 

I tested it your solution, it works perfect

**Comment by ccordoba12** [comment]
Great! Would you mind to open a pull request about it?

**Comment by Khalilsqu** [comment]
Glade to do that. Thanks a lot..

**Comment by bcolsen** [comment]
@ccordoba12 I consider coping and pasting file path from the file browser to be incredibly handy.  I do this all the time. On Linux I get a nice ready to use file path.

If the windows file paths are unusable in a python script with the `file:\\\` stuff in the front then the fix to the this issue is to make the url paste as a usable python path string like `"D:\\path\\file.ext"`.

**Comment by Khalilsqu** [comment]
@bcolsen Good point, I am not a linux user, therefore, I do not understand how exactly file system in linux works when copying from clipboard. I can address your concerns in a new PR shortly. 

I know that this behaviour does not exist in any editor I am aware of, even simple ones like notepad. windows own products, excel, word, PP does not allow pasting files naming when these files are copied into the clipboard. The same is true in VScode, Pycharm, etc.

By the way. when the PR #8494 is accepted and merged, it would be easier to copy file path than browsing OS file manager. 

Thanks for your feedback.

**Comment by Khalilsqu** [comment]
@bcolsen @ccordoba12 
The solution below works better than the previously provided solution. I am sorry about previous solution! It still prevents pasting file names with prefix `file:\\\` when these files/folders are copied into the clipboard which is the standard in most text editors in windows.

Would you mind testing it? I do Have only WSL without linux file browser to test how it works.

```
    @Slot()
    def paste(self):
        """
        Reimplement QPlainTextEdit's method to fix the following issue:
        on Windows, pasted text has only 'LF' EOL chars even if the original
        text has 'CRLF' EOL chars
        """
        clipboard = QApplication.clipboard()
        if clipboard.mimeData().hasFormat("text/plain"):
            text = to_text_string(clipboard.text())
            if len(text.splitlines()) > 1:
                eol_chars = self.get_line_separator()
                text = eol_chars.join((text + eol_chars).splitlines())
                clipboard.setText(text)
            # Standard paste
            TextEditBaseWidget.paste(self)
            self.document_did_change(text) 
```

**Comment by Khalilsqu** [comment]
Okay I just tested the first option in my previous comment in linux after installing a file browser and indeed it works. you still can paste file paths when copying them from a file manager (I tested using Nautilus) I believe other file managers in linux works similarly. The method prevents pasting file names only if copied properly from (windows file manager >> home>>copy path)..

Thanks @bcolsen very much for your feedback.
